### ISSUE TITLE
OpenCV 2.4.2

### ISSUE THREAD

**Issue by Ralvke** [lazywei/go-opencv#51]
Where get a 2.4.2 version for this binding?

**Comment by brunoqc** [comment]
I'm also wondering about this.

**Comment by lazywei** [comment]
Hi @Ralvke @brunoqc sorry for late reply and thanks for asking.

TLDL: A very basic wrapper (w/ only a few APIs wrapped) can be found [here](https://github.com/lazywei/go-opencv/tree/master/gocv)

Basically, OpenCV APIs for 2+ are only provided in C++ but no C, which means CGO will not work in such case. So I tried to utilize SWIG for wrapping C++ APIs in [gocv](https://github.com/lazywei/go-opencv/tree/master/gocv). Yet, the wrapping is still in very early stage and very few functions are provided. For now, v2+ APIs will be implemented only "on-demand", i.e., when there is a feature requesting regarding for them. Of course, PR for wrapping 2+ api is more than welcome.

Thanks.

**Comment by brunoqc** [comment]
Does [imread()](http://docs.opencv.org/modules/highgui/doc/reading_and_writing_images_and_video.html?highlight=imread#Mat imread(const string& filename, int flags)) need to be wrapped or is it already available by the Swig process? I have a project using Swig but it's still a bit confusing.

**Comment by lazywei** [comment]
it's not yet been wrapped. i'll find time to work on it asap. of course, sending a PR is also very welcome.
thanks.

**Comment by arvinasokan** [comment]
Hi there,
I am extremely new to Go and so forgive me if I am asking something primitive or dumb.
I was wondering if OpenCV had to be installed for this binding to work? 
The reason I am asking this is because I have intentions on making applications with openCV and Go which are standalone and portable in nature, like my go package should contain all the necessary dependencies.

**Comment by lazywei** [comment]
Yes, you would need to install opencv first to be able to use this wrapper
because this wrapper refers to the header files of opencv. You have to let
the compiling tools be able to find these headers.
Arvin Asokan <notifications@github.com>於 2016年1月4日 週一，上午8:12寫道：

> Hi there,
> I am extremely new to Go and so forgive me if I am asking something
> primitive or dumb.
> I was wondering if OpenCV had to be installed for this binding to work?
> The reason I am asking this is because I have intentions on making
> applications with openCV and Go which are standalone and portable in
> nature, like my go package should contain all the necessary dependencies.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/lazywei/go-opencv/issues/51#issuecomment-168557069>.
>
-- 
Chih-Wei (Bert)
